compiler/src/iree/compiler/DispatchCreation/CollapseDimensions.cpp

```diff
@@ -405,19 +408,133 @@ CollapseInfo::getTransformedCollapsableLoops(AffineMap map) const {
   return transformedLoops;
 }
 
-// Update `collapsableLoops` by taking the set intersection with
-// `otherCollapsable` and update the reassociation indicies accordingly.
-bool CollapseInfo::updateCollapseViaIntersect(
-    const CollapsableLoopsSet &otherCollapsable) {
-  CollapsableLoopsSet toRemove;
-  for (auto elem : collapsableLoops) {
-    if (!otherCollapsable.contains(elem)) {
-      toRemove.insert(elem);
+FailureOr<SmallVector<ReassociationIndices>>
+CollapseInfo::getTransformedReassociation(AffineMap map) const {
+  if (!map) {
+    return failure();
+  }
+
+  SmallVector<ReassociationIndices> transformedReassociation(reassociation);
+  for (const auto &[idxOne, indicies] : llvm::enumerate(reassociation)) {
+    for (auto [idxTwo, elem] : llvm::enumerate(indicies)) {
+      assert(elem < map.getNumResults() && "index has no valid mapping");
+      auto dimExpr = dyn_cast<AffineDimExpr>(map.getResult(elem));
+      if (!dimExpr) {
+        continue;
```

Comment:
When does it happen? Do we return a failure when it is not a `AffineDimExpr`?

Reply:
It occurs in the `quantized_matmul` testcase because the 2 reduction loops dont map to any loops in the producer elementwise op. But this needs to use push_back

Reply:
Makes sense, thanks!